packages/core/src/app/common/resolver/resolveComponent.spec.tsx

```diff
@@ -71,4 +71,56 @@ describe('resolveComponent', () => {
             Default,
         );
     });
+
+    it('returns correct component for an entry', () => {
+        const Component = toResolvableComponent(
+            ({ message }: TestingProps) => <div>Foo: {message}</div>,
+            [{ id: 'credit_card', gateway: 'bluesnap' }]
+        );
+
+        const CreditCard = resolveComponent(
+            { id: 'credit_card' },
+            { Component }
+        );
+
+        const Bluesnap = resolveComponent(
+            { id: 'credit_card' },
+            { Component }
+        );
+
+        const CheckoutCom = resolveComponent(
+            { id: 'credit_card', gateway: 'checkoutcom' },
+            { Component }
+        );
+
+        expect(CreditCard).toBeDefined();
+        expect(Bluesnap).toBeDefined();
+        expect(CheckoutCom).toBeUndefined();
+    });
+
+    it('returns correct component for an entry in paypal', () => {
```

Comment:
We can update the naming with:
`returns default component for an entry which does not have specific component to resolve` or something similar